Documentation for network headers capture (#36291)
DataDog/documentation: layouts/shortcodes/mdoc/en/sdk/advanced_config/android.mdoc.md

{% /tab %}
{% /tabs %}

### Capture resource headers

When [tracking resources automatically][6], you can capture HTTP request and response headers on RUM Resources by calling `trackResourceHeaders` on the `DatadogInterceptor.Builder`.

Captured headers appear on the RUM Resource event under `resource.request.headers` and `resource.response.headers` and are queryable in the RUM Explorer.

{% tabs %}
{% tab label="Kotlin" %}

```kotlin
val interceptor = DatadogInterceptor.Builder(tracedHosts)
    .trackResourceHeaders()
    .build()
```

{% /tab %}
{% tab label="Java" %}

```java
DatadogInterceptor interceptor = new DatadogInterceptor.Builder(tracedHosts)
    .trackResourceHeaders()
    .build();
```

{% /tab %}
{% /tabs %}

> With no arguments, a list of default headers is captured (such as `cache-control`, `content-type`, `vary`, `server-timing`, etc.).

Reviewer: Should we list all default headers? I don't believe there is any other reference to it, so customers can't know which one will be by default.

{% /tab %}
{% /tabs %}

### Capture resource headers

When [tracking resources automatically][6], you can capture HTTP request and response headers on RUM Resources by calling `trackResourceHeaders` on the `DatadogInterceptor.Builder`.

> Captured headers appear on the RUM Resource event under `resource.request.headers` and `resource.response.headers` and are queryable in the RUM Explorer.

Reviewer 2: ```suggestion
Captured headers appear on the RUM Resource event under `resource.request.headers` and `resource.response.headers`. You can query them in the RUM Explorer.
```

{% tabs %}
{% tab label="Kotlin" %}

```kotlin
val interceptor = DatadogInterceptor.Builder(tracedHosts)
    .trackResourceHeaders()
    .build()
```

{% /tab %}
{% tab label="Java" %}

```java
DatadogInterceptor interceptor = new DatadogInterceptor.Builder(tracedHosts)
    .trackResourceHeaders()
    .build();
```

{% /tab %}
{% /tabs %}

With no arguments, `trackResourceHeaders` captures a predefined set of common headers:

| Direction | Headers |
|-----------|---------|
| Request | `cache-control`, `content-type` |
| Response | `age`, `cache-control`, `content-encoding`, `content-length`, `content-type`, `etag`, `expires`, `server-timing`, `vary`, `x-cache` |

To capture additional headers on top of the defaults, configure a `ResourceHeadersExtractor` and pass it to `trackResourceHeaders`. To skip the defaults, set `includeDefaults = false`.

{% alert level="info" %}

> Sensitive headers (tokens, API keys, etc.) are filtered out automatically, even if you list them explicitly.

Reviewer 2: ```suggestion
Sensitive headers, such as tokens and API keys, are filtered out automatically, even if you list them explicitly.
```
